docs/source/frequency_guide.rst

```diff
@@ -0,0 +1,212 @@
+priorities of the user. If performance repeatability is critical or the user
+knows of ideal frequency settings, direct frequency requests are likely ideal.
+If the user wishes to leverage as much of the power headroom as possible, SST
+or HWP interfaces are useful. If the user also wishes the system to steer power
+intelligently between cores/uncore based on internal telemetry, HWP will work
+best. If computation on a given node is heterogeneous (i.e. some CPUs are given
+more work or more critical work than others), SST features are likely to work
+best.
+
+Linux Frequency Drivers
+-----------------------
+
+Linux CPU driver information can be found using ``cpupower`` as shown in the
+example below. If the driver has a ``userspace`` governor available, that will
+generally not interfere with GEOPM frequency decisions.
+
+.. code-block:: bash
+
+    $ cpupower frequency-info
+    analyzing CPU 0:
+      driver: acpi-cpufreq
+      CPUs which run at the same hardware frequency: 0
+      CPUs which need to have their frequency coordinated by software: 0
+      maximum transition latency: 10.0 us
+      hardware limits: 1000 MHz - 2.10 GHz
+      available frequency steps:  2.10 GHz, 2.10 GHz, 2.00 GHz, 1.90 GHz, 1.80 GHz, 1.70 GHz, 1.60 GHz, 1.50 GHz, 1.40 GHz, 1.30 GHz, 1.20 GHz, 1.10 GHz, 1000 MHz
+      available cpufreq governors: ondemand performance schedutil
+      current policy: frequency should be within 1000 MHz and 2.10 GHz.
+                      The governor "performance" may decide which speed to use
+                      within this range.
+      current CPU frequency: Unable to call hardware
+      current CPU frequency: 1.02 GHz (asserted by call to kernel)
+      boost state support:
+        Supported: yes
+        Active: yes
+
+When using the ``userspace`` or ``performance`` governor, the ``acpi-cpufreq``
+driver will not interfere with frequency decisions made by GEOPM. Other
+governors may interfere with GEOPM frequency settings.
+
+The ``intel_pstate`` driver allows users access to ``Hardware P-States`` on
+systems that support that feature. The ``performance`` governor is more likely
+to adhere to GEOPM frequency decisions. The ``powersave`` governor tends to
+lower CPU frequency when cores are running idle.
+
+
+Direct Frequency Requests
+-------------------------
+
+This is the most straightforward mechanism for selecting a specific frequency.
+
+
+.. code-block:: bash
+
+    $ geopmread CPU_FREQUENCY_STATUS core 0
+    3400000000
+
+    $ geopmwrite MSR::PERF_CTL:FREQ core 0 1.2e9
+    $ geopmread CPU_FREQUENCY_STATUS core 0
+    1200000000
+
+To set CPU uncore frequency, both min/max can be specified. To fix
+uncore frequency to a specific value, set ``CPU_UNCORE_FREQUENCY_MAX_CONTROL``
+and ``CPU_UNCORE_FREQUENCY_MIN_CONTROL`` to the same value. Note that there is
+no guarantee that the specified frequencies will be achieved, as the system may
+be limited by hardware constraints.
```

Comment:
Remind me: is it still necessary to be mindful of the order in which these controls are set? If so, this might be a good opportunity to make a note of that. If I remember correctly, for batch operations we ensure these are set properly.

There are multiple spots where there's reference to min and max controls, so if a note is needed about this, it might be a good idea to figure out how to mention it without having to put it in multiple places.